bootstrap/kubeadm/internal/webhooks/kubeadmconfig.go

```diff
@@ -65,6 +65,9 @@ func (webhook *KubeadmConfig) ValidateCreate(_ context.Context, obj runtime.Obje
 	if !ok {
 		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected a KubeadmConfig but got a %T", obj))
 	}
+	if c.Spec.BootCommands != nil && c.Spec.Format != bootstrapv1.CloudConfig {
```

Comment:
It seems this commands need to be a part of `webhook.validate`, otherwise the check looks fine.

Reply:
I had to update the `validateIgnition` function so that `allErrs` wouldn't be empty, which was probably for the best, thanks

https://github.com/kubernetes-sigs/cluster-api/pull/11271/commits/3db48fac929b60f078be4204d3df03211ff2e0bc

Reply:
@Danil-Grigorev Please take a look if you have time :slightly_smiling_face: